webhooks.go

```diff
@@ -0,0 +1,231 @@
+	Object string `json:"object"`
+	Id     string `json:"id"`
+}
+
+// ListWebhooksResponse represents the response from listing webhooks
+type ListWebhooksResponse struct {
+	Object  string          `json:"object"`
+	HasMore bool            `json:"has_more"`
+	Data    []WebhookInList `json:"data"`
+}
+
+// WebhookInList represents a webhook in the list response
+type WebhookInList struct {
+	Id        string   `json:"id"`
+	CreatedAt string   `json:"created_at"`
+	Status    string   `json:"status"`
+	Endpoint  string   `json:"endpoint"`
+	Events    []string `json:"events"`
+}
+
+// DeleteWebhookResponse represents the response from deleting a webhook
+type DeleteWebhookResponse struct {
+	Object  string `json:"object"`
+	Id      string `json:"id"`
+	Deleted bool   `json:"deleted"`
+}
+
+// WebhooksSvc defines the interface for webhook operations
+type WebhooksSvc interface {
+	CreateWithContext(ctx context.Context, params *CreateWebhookRequest) (*CreateWebhookResponse, error)
+	Create(params *CreateWebhookRequest) (*CreateWebhookResponse, error)
+	GetWithContext(ctx context.Context, webhookId string) (*Webhook, error)
+	Get(webhookId string) (*Webhook, error)
+	UpdateWithContext(ctx context.Context, webhookId string, params *UpdateWebhookRequest) (*UpdateWebhookResponse, error)
+	Update(webhookId string, params *UpdateWebhookRequest) (*UpdateWebhookResponse, error)
+	ListWithOptions(ctx context.Context, options *ListOptions) (*ListWebhooksResponse, error)
+	ListWithContext(ctx context.Context) (*ListWebhooksResponse, error)
+	List() (*ListWebhooksResponse, error)
+	RemoveWithContext(ctx context.Context, webhookId string) (*DeleteWebhookResponse, error)
+	Remove(webhookId string) (*DeleteWebhookResponse, error)
+}
+
+// WebhooksSvcImpl implements the WebhooksSvc interface
+type WebhooksSvcImpl struct {
+	client *Client
+}
+
+// CreateWithContext creates a new webhook with the given context
+// https://resend.com/docs/api-reference/webhooks/create-webhook
+func (s *WebhooksSvcImpl) CreateWithContext(ctx context.Context, params *CreateWebhookRequest) (*CreateWebhookResponse, error) {
+	path := "webhooks"
+
+	// Prepare request
+	req, err := s.client.NewRequest(ctx, http.MethodPost, path, params)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build response object
+	webhookResp := new(CreateWebhookResponse)
+
+	// Send Request
+	_, err = s.client.Perform(req, webhookResp)
+	if err != nil {
+		return nil, err
+	}
+
+	return webhookResp, nil
+}
+
+// Create creates a new webhook
+func (s *WebhooksSvcImpl) Create(params *CreateWebhookRequest) (*CreateWebhookResponse, error) {
+	return s.CreateWithContext(context.Background(), params)
+}
+
+// GetWithContext retrieves a webhook by ID with the given context
+// https://resend.com/docs/api-reference/webhooks/get-webhook
+func (s *WebhooksSvcImpl) GetWithContext(ctx context.Context, webhookId string) (*Webhook, error) {
+	path := "webhooks/" + webhookId
+
+	// Prepare request
```

Comment:
I'll stop commenting on these comment removals, but I feel like we could clean them up. What do you think?

```suggestion
```